app/models/preserved_copy.rb

```diff
@@ -18,6 +9,16 @@ class PreservedCopy < ApplicationRecord
     expected_version_not_found_online: 4,
     fixity_check_failed: 5
   }
+
   DEFAULT_STATUS = statuses[:ok].freeze
+
+  belongs_to :preserved_object
+  belongs_to :endpoint
+
+  validates :preserved_object, presence: true
+  validates :endpoint, presence: true
+  validates :version, presence: true
   validates :status, inclusion: { in: statuses.keys }
```

Comment:
Is it the case that we can't validate status inclusion until after the declaration?

And why validate size last?  maybe just alphabetical?

Reply:
> Is it the case that we can't validate status inclusion until after the declaration?

i can do some testing, but that'd be my assumption.  i don't believe `statuses` will exist till th enum is declared.  FWIW, an enum seems at least as fundamental to a class definition as validation, so it seems sensible to have it come first.  that's, uh, my completely objective verdict on that 🙂 

> And why validate size last? maybe just alphabetical?
almost alphabetical?  i think it was more like "scenario that needs explaining goes last".  happy to change order to alpha or whatever.